test/bench/spec/bsonBench.ts

```diff
@@ -78,6 +78,8 @@ suite.task({
 });
 
 suite.run().then(async () => {
+  if (suite.errors.length) process.exitCode = 1;
```

Comment:
Its best practice to immediately exit after setting `exitCode`.  There's nothing preventing something that runs after this from assigning exitCode to a different value.

Reply:
> Its best practice to immediately

Where are you getting this practice from?

> nothing preventing something that runs after this from assigning exitCode to a different value

That's correct, but may be desirable

Reply:
> > Its best practice to immediately

> Where are you getting this practice from? Mocha does not exit when a test fails

I don't know of anywhere that this is stated, this feels more like common sense.  Let me rephrase though, I meant mainly that if you intend to exit a process with an exit code, then you should exit immediately to ensure you exit with that intended code.

Reply:
I must lack some of that sense that is common 🙂

Node.js exits when the current exec stack is empty and there are no ref-ed resources in the event loop. When you `exit()`, you preempt remaining resources from handling their events and you cover up leaked resources that would keep Node.js running.

As a compromise, I think we should instead leave the error code setting up to Node the way we normally do by throwing an error, this also stops us from running the logic that comes after these checks to not perform unnecessary work.